src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -0,0 +1,623 @@
+    isPublishedEmailEnabled: true,
+
+    isSaving: false,
+    isEditable: false,
+    hasVisibleSettingsPanelExpanded: false,
+    hasEmailSettingsPanelExpanded: false,
+  };
+
+  // to get the original question model
+  feedbackQuestionModels: Map<string, FeedbackQuestion> = new Map();
+
+  questionEditFormModels: QuestionEditFormModel[] = [];
+
+  newQuestionEditFormModel: QuestionEditFormModel = {
+    feedbackQuestionId: '',
+    questionNumber: 0,
+    questionBrief: '',
+    questionDescription: '',
+
+    isQuestionHasResponses: false,
+
+    questionType: FeedbackQuestionType.TEXT,
+    questionDetails: {
+      recommendedLength: 0,
+    },
+
+    giverType: FeedbackParticipantType.STUDENTS,
+    recipientType: FeedbackParticipantType.STUDENTS,
+
+    numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting.UNLIMITED,
+    customNumberOfEntitiesToGiveFeedbackTo: 1,
+
+    showResponsesTo: [],
+    showGiverNameTo: [],
+    showRecipientNameTo: [],
+
+    isEditable: true,
+    isSaving: false,
+  };
+
+  isAddingQuestionPanelExpanded: boolean = false;
+
+  constructor(private route: ActivatedRoute, private router: Router, private httpRequestService: HttpRequestService,
+              private statusMessageService: StatusMessageService, private navigationService: NavigationService,
+              private timezoneService: TimezoneService, private feedbackQuestionsService: FeedbackQuestionsService) {
+  }
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((queryParams: any) => {
+      this.user = queryParams.user;
+      this.courseId = queryParams.courseid;
+      this.feedbackSessionName = queryParams.fsname;
+
+      this.loadFeedbackSession();
+      this.loadFeedbackQuestions();
+    });
+  }
+
+  /**
+   * Loads a feedback session.
+   */
+  loadFeedbackSession(): void {
+    const paramMap: { [key: string]: string } = { courseid: this.courseId, fsname: this.feedbackSessionName };
+    this.httpRequestService.get('/session', paramMap)
+        .subscribe((feedbackSession: FeedbackSession) => {
+          this.sessionEditFormModel = this.getSessionEditFormModel(feedbackSession);
+        }, (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorMessage(resp.error.message);
+        });
+  }
+
+  /**
+   * Gets the {@code sessionEditFormModel} with {@link FeedbackSession} entity.
+   */
+  getSessionEditFormModel(feedbackSession: FeedbackSession): SessionEditFormModel {
+    const submissionStart: {date: DateFormat; time: TimeFormat} =
+        this.getDateTimeAtTimezone(feedbackSession.submissionStartTimestamp, feedbackSession.timeZone);
+
+    const submissionEnd: {date: DateFormat; time: TimeFormat} =
+        this.getDateTimeAtTimezone(feedbackSession.submissionEndTimestamp, feedbackSession.timeZone);
+
+    const model: SessionEditFormModel = {
+      courseId: feedbackSession.courseId,
+      timeZone: feedbackSession.timeZone,
+      courseName: 'USE COURSE API',
```

Comment:
Let's not forget to address this.

Reply:
I thought someone would implement the course API so I can reuse?

Reply:
As it turns out, nobody has done it yet. Either that or we need to consolidate the APIs later.